### PR TITLE
Fix ci failures in 111

### PR DESCRIPTION
**Pull request by bernd-edlinger**
##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: .github/workflows/cross-compiles.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ jobs:
             tests: none
           }
         ]
-    runs-on: ubuntu-latest
```
Comment (slontis) [inline .github/workflows/cross-compiles.yml:120]: Why is this only a problem on 1.1.1? (What is the reason for the change).

Comment (bernd-edlinger) [reply .github/workflows/cross-compiles.yml:120]: the used gcc-11 version creates false-positive warnings in m68k-linux-gnu which make the build fail,
the warning is due to a large refactoring in the gcc-internal variable range computation engine ("ranger")
https://gcc.gnu.org/wiki/AndrewMacLeod/Ranger
that was rolled out in gcc-11 and gcc is still recovering from this major change.
So instead of disabling a potentially useful warning, I wanted to keep the test using ubuntu-20.04, and if that
version is no longer available, skip directly to ubuntu-24.04, which will be using gcc-13, and that warning
is already gone in gcc-trunk of today, I double-checked that.

**Comment by t8m** [comment]
OTC: Are these strict warning fixes something we want on 1.1.1 branch?

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Removing the OTC hold as we removed it from a similar (warning fix) issue on today's OTC meeting.

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
I would be OK with treating this as urgent as this is a CI fix.

**Comment by mattcaswell** [comment]
I will merge shortly

**Comment by mattcaswell** [comment]
Pushed. Thanks.
